File: 2404-most-frequent-even-element/2404-most-frequent-even-element.cpp
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
        int ans=0;
        unordered_map<int,int>ump;
        for(int &x:nums){
            if((x & 1)==0){
                ump[x]++;
                ans=max(ans,ump[x]);
            }
        }
        int res=1e6;
        for(auto [x,y]:ump){
            if(y==ans){
                res=min(res,x);
            }
        }
        if(res==1e6)return -1;
        return res;
    }
};

File: 2404-most-frequent-even-element/2404-most-frequent-even-element.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        res=-1
        fr=0
        for x in nums:
            if((x&1)==0):
                fr1=nums.count(x);
                if(fr1>fr):
                    fr=fr1
                    res=x
                elif(fr1==fr):
                    if(res>x):
                        res=x
        return res